### PR TITLE
Spi update

### PR DESCRIPTION
**Pull request by beserge**
Updates the SPI peripheral to use the pimpl format.
Also adds configuration over
- peripheral
- mode
- direction
- clock polarity
- clock phase
- nss
- baudrate prescaling
- pins

The most pressing TODO is probably implementing functions to handle RX, since we have RX configs now, but no actual functions to let you RX.

### CODE REVIEW

File: src/per/spi.cpp [hunk 1]
```diff
@@ -8,126 +11,547 @@
...
+    if(HAL_SPI_Transmit(&hspi_, buff, size, 100) != HAL_OK)
+    {
+        return SpiHandle::Result::ERR;
+    }
+    return SpiHandle::Result::OK;
 }
 
-void HAL_SPI_MspInit(SPI_HandleTypeDef* spiHandle)
+SpiHandle::Result
+SpiHandle::Impl::PollReceive(uint8_t* buffer, uint16_t size, uint32_t timeout)
```
Comment (TheSlowGrowth) [inline src/per/spi.cpp:210]: Seems like the corresponding function in the header file is mising

Comment (beserge) [reply src/per/spi.cpp:210]: yep, fixed

**Comment by stephenhensley** [comment]
Before I do the physical test for this, I figured I'd just leave a few notes on stuff I noticed looking through the files:

* The TODO at the top of the file can be updated. We should be able to get rid of "add configuration".
* The old `SpiPin` enum is [still in place](https://github.com/electro-smith/libDaisy/blob/76f44f6df9c16be4b5435b8c553f5dd6652dcba4/src/per/spi.h#L21-L28). It doesn't seem to be used anymore, I think we can get rid of it.
* `SpiHandle::Config::BaudPrescalar::BAUDRATEPRESCALER_2` is really verbose. I think we can truncate the member names to `PS_2`, `PS_4`, without it being any less clear.
    * Related, at some point I'd like to have a "target_frequency" member a real value in Hz to use instead of an arbitrary prescalar.
* I don't like the inconsistent naming between `BlockingTransmit` and `PollReceive`. For now lets make it consistent with the previously existing function. So `PollReceive`->`BlockingReceive`. And then post v0.1, we should go through and make all of the core peripherals have consistent naming for common function types.
* Related, we should update the BlockingTransmit function to have an (optional) timeout argument -- default can be 100ms since that's what's hard coded now.

Otherwise, the implementation looks solid. I'd probably mark all the `pin_alt_spi` stuff as static so that its clear that it should only ever be used in this file.

**Comment by stephenhensley** [comment]
Cool. Thanks for making the changes. Still passes the tests I ran yesterday. I think it's good to go.

**Pull request merged**